### DISCUSSION TITLE
[Ubuntu images] Why is /usr/local/bin set to 777 permission?

### DISCUSSION THREAD

**Discussion by yhaliaw** [actions/virtual-environments#7607] [General questions]
From testing in GitHub Actions, the permission for `/usr/local/bin` inside a ubuntu-22.04 runner is set to `drwxrwxrwx`.

The common permission for `/usr/local/bin` generally does not include write access for all users.
Since the runner user has sudo access, writing to `/usr/local/bin` would just require using `sudo`.

Is there any reason behind the design decision to set the permission to 777 for /usr/local/bin?

**Comment by vpolikarpov-akvelon** [comment]
Hi @yhaliaw. It was done to allow runner user to install npm modules globally without `sudo`. Here is an issue related to this: https://github.com/actions/runner-images/issues/3727.
